sketch_oct9a.ino

```diff
@@ -201,7 +180,98 @@ void loop() {
   if (alt.getStatus() == Altimeter::ACTIVE) {
     last_alt = alt.getAltitude();
   }
+}
+
+void sendDataToRadio() {
+  Frame f = createFrame(
+    last_coords,
+    last_accel,
+    last_gyro,
+    last_alt
+#if CAPSULE == 2
+    ,
+    last_voc,
+    last_humid,
+    last_temp
+#endif
+  );
+
+  // In theory, this should send the data to the radio.
+  // In practice, this is completely untested. If nothing is connected to the UART pins, however,
+  // this doesn't do anything, so it's harmless to leave in for now.
+  Serial1.write(
+    (char*)&f,
+    FRAME_SIZE()
+  );
+}
+
+void setup() {
+  Serial1.begin(230400);
 
+#if CAPSULE == 2
+  // Pin A6: analog input from VOC sensor
+  pinMode(A6, PinMode::INPUT);
+#endif
+  // Pin D5: SD LEDs
+  pinMode(5, PinMode::OUTPUT);
+  // Pin D6: GPS LEDs
+  pinMode(6, PinMode::OUTPUT);
+  // Pin D7: sensor LEDs
+  pinMode(7, PinMode::OUTPUT);
+
+  pinPeripheral(0, PIO_SERCOM);
+  pinPeripheral(1, PIO_SERCOM);
+  pinPeripheral(8, PIO_SERCOM);
+  pinPeripheral(9, PIO_SERCOM);
+  pinPeripheral(A1, PIO_SERCOM);
+  pinPeripheral(A2, PIO_SERCOM);
+  pinPeripheral(A3, PIO_SERCOM);
+  pinPeripheral(A4, PIO_SERCOM);
+
+  // Turn on the error LEDs until initialization finishes
+  digitalWrite(5, LOW);
+  digitalWrite(6, LOW);
+  digitalWrite(7, LOW);
+
+  // Wait for a command from the radio. Sometimes the sensors need multiple tries, and calling
+  // initialize too many times does nothing, so put the initialize call in the loop.
+  do {
+    initializeAll();
+
+    if (Serial1.available()) {
```

Comment:
why is it 'Serial1' ? There doesn't seem to be any 'Serial0' or 'Serial2'. I'm assuming this serial object is dealing with the radio/capsule interface, or is it some other interface?

Reply:
On the MKR Zero specifically, `Serial` is the USB port and `Serial1` is the UART pins.